web/src/main/SigninBox.js

```diff
@@ -0,0 +1,245 @@
+// Copyright 2021 The casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// restored from commit 9f260a7
+// modified to adapt Casdoor JavaScript SDK
+
+import React from "react";
+import Header from "./Header";
+import * as AccountBackend from "../backend/AccountBackend";
+import * as Setting from "../Setting";
+import { withRouter, Link } from "react-router-dom";
+import i18next from "i18next";
+import { Casdoor } from "../Casdoor/Casdoor";
+import { AuthConfig } from "../Conf";
+
+class SigninBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classes: props,
+      form: {},
+      isTypingStarted: false,
+      problem: [],
+      message: "",
+    };
+
+    let url = `${window.location.protocol}//${window.location.hostname}`;
+    let port = window.location.port;
+    if (port !== "") url = `${url}:${port}`;
+    url += "/callback";
+
+    this.Casdoor = new Casdoor(
```

Comment:
The `casdoor` instance will be frequently used in various components, would it be better to initialize in `App.js`?

Reply:
That's right, I will move the Casdoor instance to App.js.

Reply:
@kocoler Moved to `App.js`